### DISCUSSION TITLE
Mapping over an list?

### DISCUSSION THREAD

**Discussion by mon-jai** [nushell/nushell#7678] [Ideas]
In JavaScript or Python, we can create a new array using elements from old array.

```js
const arr1 = [1, 2, 3]
const arr2 = [1, 2, 3].map(i => i + 1) // [2, 3, 4]
```

```python
arr1 = [1, 2, 3]
arr2 = [i + 1 for i in arr1] # [2, 3, 4]
```

Does Nushell support similar operations over a list?

**Comment by fdncred** [comment]
We have `each` and `reduce` that iterate through lists. You can see more with `help each` and `help reduce`. I don't think we have an exact mapping for `map` though.

**Comment by mon-jai** [comment]
Would you consider supporting map operations?

It will be quite useful when dealing with data, and also making scripts easier to read.

**Reply by fdncred** [reply]
ya, we'd love to have someone contribute a `map` command.

**Reply by rgwood** [reply]
IMO `each` is Nushell's equivalent of `map`. It's not clear to me what a new `map` command would do differently.

If you prefer the name `map` to `each`, an alias might help: `alias map = each`

**Reply by mon-jai** [reply]
@rgwood That is what I wanted.

It is not obvious though. Maybe we can mention it in the [list section of documentations](https://www.nushell.sh/book/working_with_lists.html).

**Comment by TJTorola** [comment]
Just spent far too long looking for something like a `map` function. Arguably my own fault for not reading the documentation for `each` closely enough but I ended up assuming that `each` was for side effects of some sort and not returning new values for a list.

**Reply by ghost** [reply]
a tip for the future: `help --find map`.  
if other languages call a command something else it is often marked as a search-term and thus will show up.

**Reply by fdncred** [reply]
There's also the documentation "Coming to Nu" section that could be helpful like https://www.nushell.sh/book/nushell_map_functional.html

**Reply by TJTorola** [reply]
Thanks for the tips. I haven't really been using the builtin `help` I'll try to use that more going forward.
